com.hazelcast.instance.impl.HazelcastInstanceFactoryTest.test_NewInstance_failed_beforeNodeShutdown

_master_ (commit 333a1b9168df26f7fdcb95b833f2b417f3e4dbca)

Failed on OracleJDK 8 with nightly profile: http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-OracleJDK8-nightly/845/testReport/com.hazelcast.instance.impl/HazelcastInstanceFactoryTest/test_NewInstance_failed_beforeNodeShutdown/

Stacktrace:
```
java.lang.AssertionError: Expected exception: com.hazelcast.test.ExpectedRuntimeException
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:34)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

Standard output:
```
03:50:23,082 DEBUG |testTestHazelcastInstanceFactory_withTwoFactories| - [JobCoordinationService] hz.heuristic_germain.cached.thread-86 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:22,632 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-147 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:23,180  INFO |testTestHazelcastInstanceFactory_withTwoFactories| - [HealthMonitor] hz.heuristic_germain.HealthMonitor - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] processors=72, physical.memory.total=377.6G, physical.memory.free=147.7G, swap.space.total=4.0G, swap.space.free=2.0G, heap.memory.used=1.2G, heap.memory.free=541.2M, heap.memory.total=1.8G, heap.memory.max=1.8G, heap.memory.used/total=69.80%, heap.memory.used/max=68.71%, minor.gc.count=286, minor.gc.time=68237ms, major.gc.count=26, major.gc.time=27726ms, load.process=3.70%, load.system=100.00%, load.systemAverage=6038.33, thread.count=1708, thread.peakCount=3370, cluster.timeDiff=0, event.q.size=0, executor.q.async.size=0, executor.q.client.size=0, executor.q.client.query.size=0, executor.q.client.blocking.size=0, executor.q.query.size=0, executor.q.scheduled.size=0, executor.q.io.size=0, executor.q.system.size=0, executor.q.operations.size=0, executor.q.priorityOperation.size=0, operations.completed.count=10, executor.q.mapLoad.size=0, executor.q.mapLoadAllKeys.size=0, executor.q.cluster.size=0, executor.q.response.size=0, operations.running.count=0, operations.pending.invocations.percentage=0.00%, operations.pending.invocations.count=0, proxy.count=0, clientEndpoint.count=0, connection.active.count=0, client.connection.count=0, connection.count=0
03:50:23,180  WARN |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-147 - Logging took 548 ms.
Finished Running Test: fixedNameGeneratedIfPropertyDisabled in 25.452 seconds.
Hiccups measured while running test 'test_NewInstance_failed_beforeNodeShutdown(com.hazelcast.instance.impl.HazelcastInstanceFactoryTest):'
03:49:55, accumulated pauses: 1896 ms, max pause: 266 ms, pauses over 1000 ms: 0
03:50:00, accumulated pauses: 3414 ms, max pause: 2047 ms, pauses over 1000 ms: 1
03:50:05, accumulated pauses: 1482 ms, max pause: 1067 ms, pauses over 1000 ms: 1
03:50:10, accumulated pauses: 1319 ms, max pause: 910 ms, pauses over 1000 ms: 0
03:50:15, accumulated pauses: 1817 ms, max pause: 788 ms, pauses over 1000 ms: 0
03:50:20, accumulated pauses: 1109 ms, max pause: 555 ms, pauses over 1000 ms: 0


No metrics recorded during the test
03:50:22,609  INFO |fixedNameGeneratedIfPropertyNotDefined| - [NodeExtension] Thread-572 - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] Destroying node NodeExtension.
03:50:23,189  WARN |fixedNameGeneratedIfPropertyNotDefined| - [NodeExtension] Thread-572 - Logging took 580 ms.
03:50:22,530 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-26 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:23,196  WARN |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-26 - Logging took 665 ms.
03:50:23,201  WARN |when_jetClientCreated_then_connectsToJetCluster| - [ClientConnectionManager] hz.client_66.internal-3 - hz.client_66 [a3335479-04d8-4661-9047-9ac841387c93] [5.0-SNAPSHOT] [5.0-SNAPSHOT] Exception during initial connection to Member [127.0.0.1]:5702 - a15918c2-d0c3-433a-8866-f4e4a0ab693f: com.hazelcast.core.HazelcastException: java.io.IOException: Connection refused to address /127.0.0.1:5702
03:50:23,201  INFO |when_jetClientCreated_then_connectsToJetCluster| - [ClientConnectionManager] hz.client_66.internal-3 - hz.client_66 [a3335479-04d8-4661-9047-9ac841387c93] [5.0-SNAPSHOT] [5.0-SNAPSHOT] Trying to connect to Member [127.0.0.1]:5701 - 3472b278-3823-40bd-bcc2-5b1fbf84b80f
03:50:23,202  WARN |when_jetClientCreated_then_connectsToJetCluster| - [ClientConnectionManager] hz.client_66.internal-3 - hz.client_66 [a3335479-04d8-4661-9047-9ac841387c93] [5.0-SNAPSHOT] [5.0-SNAPSHOT] Exception during initial connection to Member [127.0.0.1]:5701 - 3472b278-3823-40bd-bcc2-5b1fbf84b80f: com.hazelcast.core.HazelcastException: java.io.IOException: Connection refused to address /127.0.0.1:5701
03:50:23,202  WARN |when_jetClientCreated_then_connectsToJetCluster| - [ClientConnectionManager] hz.client_66.internal-3 - hz.client_66 [a3335479-04d8-4661-9047-9ac841387c93] [5.0-SNAPSHOT] [5.0-SNAPSHOT] Unable to get live cluster connection, retry in 5747 ms, attempt: 37, cluster connect timeout: INFINITE, max backoff: 30000 ms
03:50:23,238 DEBUG |testTestHazelcastInstanceFactory_withTwoFactories| - [JobCoordinationService] hz.heuristic_germain.cached.thread-94 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:23,284 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-19 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:23,302 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-19 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:23,386  WARN |when_jetClientCreated_then_connectsToJetCluster| - [ClientConnectionManager] hz.client_66.internal-2 - hz.client_66 [a3335479-04d8-4661-9047-9ac841387c93] [5.0-SNAPSHOT] [5.0-SNAPSHOT] Could not connect to member 3472b278-3823-40bd-bcc2-5b1fbf84b80f, reason com.hazelcast.core.HazelcastException: java.io.IOException: Connection refused to address /127.0.0.1:5701
03:50:23,387  WARN |when_jetClientCreated_then_connectsToJetCluster| - [ClientConnectionManager] hz.client_66.internal-2 - hz.client_66 [a3335479-04d8-4661-9047-9ac841387c93] [5.0-SNAPSHOT] [5.0-SNAPSHOT] Could not connect to member a15918c2-d0c3-433a-8866-f4e4a0ab693f, reason com.hazelcast.core.HazelcastException: java.io.IOException: Connection refused to address /127.0.0.1:5702
03:50:23,394 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-19 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:23,419 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-19 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:23,500 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-19 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:23,520 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-19 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:23,604 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-26 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:23,621 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-129 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:23,706 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-16 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:23,712  INFO |testTestHazelcastInstanceFactory_withTwoFactories| - [JetService] testTestHazelcastInstanceFactory_withTwoFactories - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] Setting number of cooperative threads and default parallelism to 72
03:50:23,726 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-16 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:23,808 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-65 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:23,830 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-65 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:23,909 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-65 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:23,931 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-16 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:24,011 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-16 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:24,034 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-16 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:24,116 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-129 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:24,141 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-19 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:24,219 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-19 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:24,241 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-16 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:24,321 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-16 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:24,343 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-19 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:24,366  INFO |testTestHazelcastInstanceFactory_withTwoFactories| - [Diagnostics] testTestHazelcastInstanceFactory_withTwoFactories - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
03:50:24,367  INFO |testTestHazelcastInstanceFactory_withTwoFactories| - [LifecycleService] testTestHazelcastInstanceFactory_withTwoFactories - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] [127.0.0.1]:5703 is STARTING
03:50:24,367  INFO |testTestHazelcastInstanceFactory_withTwoFactories| - [MockServer] testTestHazelcastInstanceFactory_withTwoFactories - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] Created connection to endpoint: [127.0.0.1]:5701, connection: MockConnection{localEndpoint=[127.0.0.1]:5703, remoteEndpoint=[127.0.0.1]:5701, alive=true}
03:50:24,387  WARN |when_jetClientCreated_then_connectsToJetCluster| - [ClientConnectionManager] hz.client_66.internal-2 - hz.client_66 [a3335479-04d8-4661-9047-9ac841387c93] [5.0-SNAPSHOT] [5.0-SNAPSHOT] Could not connect to member 3472b278-3823-40bd-bcc2-5b1fbf84b80f, reason com.hazelcast.core.HazelcastException: java.io.IOException: Connection refused to address /127.0.0.1:5701
03:50:24,387  WARN |when_jetClientCreated_then_connectsToJetCluster| - [ClientConnectionManager] hz.client_66.internal-9 - hz.client_66 [a3335479-04d8-4661-9047-9ac841387c93] [5.0-SNAPSHOT] [5.0-SNAPSHOT] Could not connect to member a15918c2-d0c3-433a-8866-f4e4a0ab693f, reason com.hazelcast.core.HazelcastException: java.io.IOException: Connection refused to address /127.0.0.1:5702
03:50:24,389  INFO |testTestHazelcastInstanceFactory_withTwoFactories| - [MockServer] hz.heuristic_germain.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Created connection to endpoint: [127.0.0.1]:5703, connection: MockConnection{localEndpoint=[127.0.0.1]:5701, remoteEndpoint=[127.0.0.1]:5703, alive=true}
03:50:24,396  INFO |testTestHazelcastInstanceFactory_withTwoFactories| - [ClusterService] hz.heuristic_germain.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] 

Members {size:3, ver:3} [
	Member [127.0.0.1]:5701 - 6bc665e4-b279-46b2-bbf3-31ecbe434c7c this
	Member [127.0.0.1]:5702 - 92355e23-b478-407e-8324-fcc30d28f4bd
	Member [127.0.0.1]:5703 - f5a98070-7c20-4569-836c-fbcb711b5756
]

03:50:24,396  INFO |testTestHazelcastInstanceFactory_withTwoFactories| - [MockServer] hz.hardcore_germain.priority-generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] Created connection to endpoint: [127.0.0.1]:5703, connection: MockConnection{localEndpoint=[127.0.0.1]:5702, remoteEndpoint=[127.0.0.1]:5703, alive=true}
03:50:24,397  INFO |testTestHazelcastInstanceFactory_withTwoFactories| - [ClusterService] hz.hardcore_germain.priority-generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] 

Members {size:3, ver:3} [
	Member [127.0.0.1]:5701 - 6bc665e4-b279-46b2-bbf3-31ecbe434c7c
	Member [127.0.0.1]:5702 - 92355e23-b478-407e-8324-fcc30d28f4bd this
	Member [127.0.0.1]:5703 - f5a98070-7c20-4569-836c-fbcb711b5756
]

03:50:24,431 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-16 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:24,445 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-19 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:24,490  INFO |testTestHazelcastInstanceFactory_withTwoFactories| - [JetExtension] hz.recursing_germain.priority-generic-operation.thread-0 - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] Jet extension is enabled after the cluster version upgrade.
03:50:24,491  INFO |testTestHazelcastInstanceFactory_withTwoFactories| - [ClusterService] hz.recursing_germain.priority-generic-operation.thread-0 - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] 

Members {size:3, ver:3} [
	Member [127.0.0.1]:5701 - 6bc665e4-b279-46b2-bbf3-31ecbe434c7c
	Member [127.0.0.1]:5702 - 92355e23-b478-407e-8324-fcc30d28f4bd
	Member [127.0.0.1]:5703 - f5a98070-7c20-4569-836c-fbcb711b5756 this
]

03:50:24,491  INFO |testTestHazelcastInstanceFactory_withTwoFactories| - [MockServer] testTestHazelcastInstanceFactory_withTwoFactories - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] Created connection to endpoint: [127.0.0.1]:5702, connection: MockConnection{localEndpoint=[127.0.0.1]:5703, remoteEndpoint=[127.0.0.1]:5702, alive=true}
03:50:24,491  INFO |testTestHazelcastInstanceFactory_withTwoFactories| - [JetExtension] testTestHazelcastInstanceFactory_withTwoFactories - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] Jet extension is enabled
03:50:24,495  INFO |testTestHazelcastInstanceFactory_withTwoFactories| - [LifecycleService] testTestHazelcastInstanceFactory_withTwoFactories - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] [127.0.0.1]:5703 is STARTED
03:50:24,497  INFO |testTestHazelcastInstanceFactory_withTwoFactories| - [AbstractConfigLocator] testTestHazelcastInstanceFactory_withTwoFactories - Loading 'hazelcast-default.xml' from the classpath.
03:50:24,540  INFO |testTestHazelcastInstanceFactory_withTwoFactories| - [HealthMonitor] hz.recursing_germain.HealthMonitor - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] processors=72, physical.memory.total=377.6G, physical.memory.free=147.6G, swap.space.total=4.0G, swap.space.free=2.0G, heap.memory.used=1.2G, heap.memory.free=520.4M, heap.memory.total=1.8G, heap.memory.max=1.8G, heap.memory.used/total=70.96%, heap.memory.used/max=69.85%, minor.gc.count=286, minor.gc.time=68237ms, major.gc.count=26, major.gc.time=27726ms, load.process=0.00%, load.system=100.00%, load.systemAverage=6038.33, thread.count=1891, thread.peakCount=3370, cluster.timeDiff=-1, event.q.size=0, executor.q.async.size=0, executor.q.client.size=0, executor.q.client.query.size=0, executor.q.client.blocking.size=0, executor.q.query.size=0, executor.q.scheduled.size=0, executor.q.io.size=0, executor.q.system.size=0, executor.q.operations.size=11, executor.q.priorityOperation.size=0, operations.completed.count=16, executor.q.mapLoad.size=0, executor.q.mapLoadAllKeys.size=0, executor.q.cluster.size=0, executor.q.response.size=0, operations.running.count=0, operations.pending.invocations.percentage=0.00%, operations.pending.invocations.count=0, proxy.count=0, clientEndpoint.count=0, connection.active.count=0, client.connection.count=0, connection.count=0
03:50:24,543 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-65 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:24,547 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-65 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:24,648 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-16 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:24,649 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-16 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:24,655  INFO |testTestHazelcastInstanceFactory_withTwoFactories| - [MetricsConfigHelper] testTestHazelcastInstanceFactory_withTwoFactories - [LOCAL] [dev] [5.0-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
03:50:24,657  INFO |testTestHazelcastInstanceFactory_withTwoFactories| - [system] testTestHazelcastInstanceFactory_withTwoFactories - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] 
	o  o   O   o---o o--o o      o-o   O    o-o  o-O-o     o--o  o       O  o-O-o o--o  o-o  o--o  o   o 
	|  |  / \     /  |    |     /     / \  |       |       |   | |      / \   |   |    o   o |   | |\ /| 
	O--O o---o  -O-  O-o  |    O     o---o  o-o    |       O--o  |     o---o  |   O-o  |   | O-Oo  | O | 
	|  | |   |  /    |    |     \    |   |     |   |       |     |     |   |  |   |    o   o |  \  |   | 
	o  o o   o o---o o--o O---o  o-o o   o o--o    o       o     O---o o   o  o   o     o-o  o   o o   o
03:50:24,657  INFO |testTestHazelcastInstanceFactory_withTwoFactories| - [system] testTestHazelcastInstanceFactory_withTwoFactories - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
03:50:24,657  INFO |testTestHazelcastInstanceFactory_withTwoFactories| - [system] testTestHazelcastInstanceFactory_withTwoFactories - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Hazelcast Platform 5.0-SNAPSHOT (20210519 - 333a1b9) starting at [127.0.0.1]:5701
03:50:24,657  INFO |testTestHazelcastInstanceFactory_withTwoFactories| - [system] testTestHazelcastInstanceFactory_withTwoFactories - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Cluster name: dev
03:50:24,665  INFO |testTestHazelcastInstanceFactory_withTwoFactories| - [MetricsConfigHelper] testTestHazelcastInstanceFactory_withTwoFactories - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
03:50:24,750 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-16 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
03:50:24,750 DEBUG |testTestHazelcastInstanceFactory| - [JobCoordinationService] hz.focused_germain.cached.thread-16 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
```

--- a/hazelcast/src/test/java/com/hazelcast/instance/impl/HazelcastInstanceFactoryTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/instance/impl/HazelcastInstanceFactoryTest.java
@@ -183,7 +183,7 @@ public class HazelcastInstanceFactoryTest extends HazelcastTestSupport {
                         }
                         return null;
                     }
-                }).when(nodeExtension).beforeShutdown(false);
+                }).when(nodeExtension).beforeShutdown(true);
                 return nodeExtension;
             }
         };
@@ -207,12 +207,9 @@ public class HazelcastInstanceFactoryTest extends HazelcastTestSupport {
             public NodeExtension createNodeExtension(final Node node) {
                 NodeExtension nodeExtension = super.createNodeExtension(node);
 
-                doAnswer(new Answer() {
-                    @Override
-                    public Object answer(InvocationOnMock invocation) throws Throwable {
-                        node.hazelcastInstance.shutdown();
-                        return null;
-                    }
+                doAnswer(invocation -> {
+                    node.hazelcastInstance.shutdown();
+                    return null;
                 }).when(nodeExtension).afterStart();
 
                 return nodeExtension;

